tests/hrex/test_hrex.py

```diff
@@ -40,3 +44,105 @@ def test_get_samples_by_iter_by_replica_invalid_args():
         get_samples_by_iter_by_replica([[1], [2, 3]], [[ReplicaIdx(0)], [ReplicaIdx(0)]])
     with pytest.raises(AssertionError):
         get_samples_by_iter_by_replica([[1, 2], [3, 4]], [[ReplicaIdx(0)], [ReplicaIdx(0), ReplicaIdx(1)]])
+
+
+def simulate_perfect_mixing_hrex(num_states: int, num_frames: int) -> List[List[int]]:
+    """assume every step of HREX perfectly mixed all replicas"""
+    rng = np.random.default_rng(num_states)
+    inds = np.arange(num_states)
+    traj = []
+    for _ in range(num_frames):
+        rng.shuffle(inds)
+        traj.append(np.array(inds).tolist())
+    return traj
+
+
+def simulate_slow_mixing_hrex(num_states: int, num_frames: int) -> List[List[int]]:
+    """assume every step of HREX only succeeds in making K nearest-neighbor swaps"""
```

Comment:
`K` -> `num_states`

(docstring matched signature `(K, T)`, signature now `(num_states, num_frames)`)

Reply:
Fixed up in https://github.com/proteneer/timemachine/pull/1314/commits/68079aab2b6b8fdcef0f050a75313708516ba6a3